### ISSUE TITLE
Sponsor shirt size becomes unset

### ISSUE THREAD

**Issue by jmdawson** [MidwestFurryFandom/rams#140]
When a sponsor or shiny sponsor registers, their selected shirt size becomes unset when they transition from "unpaid" to "created."

 `Attendee	unpaid preregistration	17:43 on May 15 2017	Test Developer	reg_station=None, admin_notes='', address2='', country='United States', city='il', ec_phone='3334445656', age_group='age unknown', email='crimson@vt.edu', found_how='', birthdate=datetime.date(1919, 9, 9), nonshift_hours=0, can_work_teardown=False, affiliate='', id='a898915d-df39-4e63-98c4-928c8720efd2', amount_extra='No thanks', international=False, badge_num=None, comped_reason='', requested_depts='', times_printed=0, assigned_depts='', got_merch=False, address1='il', extra_merch='', payment_method=None, print_pending=False, last_name='asdlkjasdf', amount_paid=0, badge_type='Sponsor', badge_status='New', for_review='', trusted_depts='', ec_name='asdfas', zip_code='60148', legal_name='', no_cellphone=False, checked_in=None, paid='no', can_work_setup=False, group_id=None, shirt='Medium', interests='', can_spam=True, public_id='d9259e25-48e9-41e3-981d-ebf382a6c0ae', ribbon='no ribbon', amount_refunded=0, past_years='', first_name='dasdfasdf', badge_printed_name='asdfasf', watchlist_id=None, region='Illinois', overridden_price=None, cellphone='2228884949', registered=None, placeholder=False, staffing=False, comments='', regdesk_info=''`

`Attendee	created	17:43 on May 15 2017	Test Developer	reg_station=None, admin_notes='', address2='', country='United States', city='il', ec_phone='3334445656', age_group='21 or older', email='crimson@vt.edu', found_how='', birthdate=datetime.date(1919, 9, 9), nonshift_hours=0, can_work_teardown=False, affiliate='', id='a898915d-df39-4e63-98c4-928c8720efd2', amount_extra='No thanks', international=False, badge_num=1020, comped_reason='', requested_depts='', times_printed=0, assigned_depts='', got_merch=False, address1='il', extra_merch='', payment_method=None, print_pending=True, last_name='Asdlkjasdf', amount_paid=150, badge_type='Sponsor', badge_status='Complete', for_review='', trusted_depts='', ec_name='asdfas', zip_code='60148', legal_name='', no_cellphone=False, checked_in=None, paid='yes', can_work_setup=False, group_id=None, shirt='no shirt', interests='', can_spam=False, public_id='d9259e25-48e9-41e3-981d-ebf382a6c0ae', ribbon='no ribbon', amount_refunded=0, past_years='', first_name='Dasdfasdf', badge_printed_name='asdfasf', watchlist_id=None, region='Illinois', overridden_price=None, cellphone='2228884949', registered=datetime.datetime(2017, 5, 15, 22, 43, 49, 35626, tzinfo=<UTC>), placeholder=False, staffing=False, comments='', regdesk_info=''`

**Comment by ianlayton** [comment]
Also noticed there is no way to change the shirt size on the admin side for the person. That *may* be due to the fact that it got unset possibly?

**Comment by jmdawson** [comment]
I think the issues are related.

**Comment by kitsuta** [comment]
Whoops! This was never caught in testing because we added shirt size config so late.

I pointed @jmdawson to the relevant function and he fixed it in https://github.com/MidwestFurryFandom/mff-rams-plugin/pull/13. I believe that fix has been deployed to prod. If the shirt dropdown still isn't showing up on the admin page, then that's likely a JS bug/oversight.

Until we get the JS bug fixed (all non-critical fixes are awaiting a contract) admins can use the "register themself" link to go to the attendee's confirmation page and edit the t-shirt size there.
